src/main/java/drbrown/utils/Parser.java

```diff
@@ -0,0 +1,130 @@
+package drbrown.utils;
+
+import drbrown.command.*;
+import drbrown.task.Deadline;
+import drbrown.task.Event;
+import drbrown.task.Task;
+import drbrown.task.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * Parses user input and returns the appropriate command for execution.
+ * Handles parsing for various commands such as adding tasks, marking tasks,
+ * unmarking tasks, listing tasks, deleting tasks, and exiting the application.
+ */
+public class Parser {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
```

Comment:
Good use of constant for formatter! However, maybe can you capitalized naming instead?
```suggestion
    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
```